pkg/apiserver/handlers.go

```diff
@@ -64,8 +64,8 @@ func ReadOnly(handler http.Handler) http.Handler {
 			handler.ServeHTTP(w, req)
 			return
 		}
-		w.WriteHeader(http.StatusForbidden)
-		fmt.Fprintf(w, "This is a read-only endpoint.")
+		w.Header().Set("Allow", "GET")
+		http.Error(w, "This is a read-only endpoint.", http.StatusMethodNotAllowed)
```

Comment:
Should use NewMethodNotSupported:
https://github.com/GoogleCloudPlatform/kubernetes/blob/a927c239fe5845ce2b515b932f5d7dfa7a9c88d5/pkg/api/errors/errors.go#L197

We want to return Status json structs for all errors.